src/kpi_rover_ecu/src/motor.cpp

```diff
@@ -41,9 +58,39 @@ int Motor::MotorStop() const {
     return 0;
 }
 
-int Motor::GetEncoderCounter() const {
+float Motor::GetTimeSegment() {
+    const auto kCurrentTimePoint = std::chrono::high_resolution_clock::now();
+    const std::chrono::duration<double, std::milli> kElapsedMilliseconds = kCurrentTimePoint - lastTimePoint_;
+    lastTimePoint_ = kCurrentTimePoint;
+    return static_cast<float>(kElapsedMilliseconds.count());
+}
+
+float Motor::GetSpeedError(int _ticks, float _timeSegment) {
+    const float kRevolutions = static_cast<float>(_ticks) / static_cast<float>(kLoopTicks);
+    const float kInputPoint =
+        static_cast<float>(std::round((kRevolutions * kSecondsMinute * kMiliSecondsSeconds) / _timeSegment)) *
+        kSpeedIndexMultipler;
+    actualRpm_ = static_cast<int>(kInputPoint);
+    const float kError = static_cast<float>(setpointRpm_) - kInputPoint;
+    std::cout << "set point " << setpointRpm_ << " current point " << kInputPoint << " error " << kError << '\n';
+
+    return kError;
+}
```

Comment:
This function taking two responsibilities:
1. Calculating actualRpm_
2. Calculating error.

It must be two separate functions.